### DISCUSSION TITLE
Adding Dispatch Name Column

### DISCUSSION THREAD

**Discussion by mpvgithub** [AgnostiqHQ/covalent#1070] [General]
**Description**
A feature that lets end users able to edit the name of the dispatch via the UI on the frontend. Since we are implementing sub-lattices feature, the naming will be easily identified between normal lattice and sub lattice.

**Impacted components**
1. Backend APIs and models of the web app (maintained by Team Psi)
2. Covalent data store models (maintained by Team AQ)
3. Alembic revisions (maintained by Team AQ, Team Psi will / can take over generating the revision files)
4. Front-End / UI component modifications for accepting user input

**Impacted files**

1.  `covalent/_data_store/models.py`
a. To modify for adding the new column
b. Alembic configuration will pick the db reference from here

2. `covalent/_results_manager/result.py`
a. Modification to Json data before sending to the database

3. `covalent/_results_manager/write_result_to_db.py`
a. save and commit the data to db

4. `covalent_ui/api/v1/database/schema/lattices.py`
a. Contains the db schema, similar to that in covalent/data_store/models.py
b. Adding a column, will required to be modified in both lattices.py ( managed by Team Psi ) as well as data_store/models.py ( managed by Team AQ )

5. `covalent_ui/api/v1/database/models/lattices_model.py`
a. Serves a query model to the lattice data layer

6. `covalent_ui/api/v1/routes/endpoints/lattice_route.py`
a. Add new API(s) to handle user input for dispatch_name
b. Existing API(s) to handle the modification of the schema and model for the new column ( dispatch_name )

**Roadmap for backend component**

1. Once this is stable, data_store/models and covalent_ui/models can be suggested to be merged and we can maintain a shared code base between AQ and Psiog

**Pending decisions**
Can we prefix the dispatch_name column for all the sub-lattices with the word, ":sublattice"? For instance, ":sublattice-get_info"

**Comment by mpvgithub** [comment]
@wjcunningham7 

CC: @Prasy12 @mshkanth

**Reply by wjcunningham7** [reply]
@mshkanth would it be possible to delay implementation of this until we enhance the API layer, so that fewer components will be impacted?  This way both the front- and back-end can simply query a display name from an API.

**Reply by mshkanth** [reply]
@wjcunningham7 - You are proposing that such a dispatch name feature be owned by the dispatcher and that an API to retreive/write the dispatch name would be exposed as a dispatcher API and not a webapp API. Right?
